app/services/WhatYouOweService.scala

```diff
@@ -79,18 +80,23 @@ class WhatYouOweService @Inject()(val financialDetailsService: FinancialDetailsS
           chargesList = getFilteredChargesList(financialDetailsModelList, isReviewAndReconciledEnabled, isFilterCodedOutPoasEnabled),
           codedOutDocumentDetail = codedOutChargeItem)
 
-        callOutstandingCharges(mtdUser.saUtr, mtdUser.incomeSources.yearOfMigration, dateService.getCurrentTaxYearEnd).map {
-          case Some(outstandingChargesModel) => whatYouOweChargesList.copy(outstandingChargesModel = Some(outstandingChargesModel))
-          case _ => whatYouOweChargesList
+        (mtdUser.saUtr, mtdUser.incomeSources.yearOfMigration) match {
+          case (Some(_), Some(year)) =>
+            callOutstandingCharges(dateService.getCurrentTaxYear, year).map {
+              case Some(outstandingChargesModel) => whatYouOweChargesList.copy(outstandingChargesModel = Some(outstandingChargesModel))
+              case _ => whatYouOweChargesList
+            }
+          case _ => Future.successful(whatYouOweChargesList)
         }
     }
   }
 
-  private def callOutstandingCharges(saUtr: Option[String], yearOfMigration: Option[String], currentTaxYear: Int)
-                                    (implicit headerCarrier: HeaderCarrier): Future[Option[OutstandingChargesModel]] = {
-    if (saUtr.isDefined && yearOfMigration.isDefined && yearOfMigration.get.toInt >= currentTaxYear - 1) {
-      val saPreviousYear = yearOfMigration.get.toInt - 1
-      outstandingChargesConnector.getOutstandingCharges("utr", saUtr.get, saPreviousYear.toString) map {
+  private def callOutstandingCharges(currentTaxYear: TaxYear, yearOfMigration: String)
+                                    (implicit headerCarrier: HeaderCarrier,mtdUser: MtdItUser[_]):Future[Option[OutstandingChargesModel]] = {
+
+    if (yearOfMigration.toInt >= currentTaxYear.startYear) {
+      val saPreviousYear = mtdUser.incomeSources.yearOfMigration.get.toInt - 1
```

Comment:
this data extraction looks redundant as we can simply use yearOfMigration variable already in scope:
ie => `(yearOfMigration.toInt - 1).toString`